community/primitive-collections/src/main/java/org/neo4j/concurrent/RecentK.java

```diff
@@ -30,116 +33,101 @@
  */
 public class RecentK<Type> implements Iterable<Type>
 {
+    /** Cut a number in half. */
+    public static final LongUnaryOperator COUNTER_HALFLIFE = new LongUnaryOperator()
```

Comment:
Hah, apparently it didn't matter in any case - the API I was using does not exist in java 7, hence the compile error. Replaced it with basically the same thing but without the lambda.